railties/test/railties/http_request_test.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require "isolation/abstract_unit"
+require "stringio"
+require "rack/test"
+
+module RailtiesTest
+  class HttpRequestTest < ActiveSupport::TestCase
+    include ActiveSupport::Testing::Isolation
+    include Rack::Test::Methods
+
+    def setup
+      build_app
+
+      app_file "config/routes.rb", <<-RUBY
+        Rails.application.routes.draw do
+          post 'posts', :to => 'posts#create'
+        end
+      RUBY
+
+      controller "posts", <<-RUBY
+        class PostsController < ApplicationController
+          def create
+            render plain: request.raw_post
+          end
+        end
+      RUBY
+    end
+
+    def teardown
+      teardown_app
+    end
+
+    def boot_rails
+      require "#{app_path}/config/environment"
+    end
```

Comment:
I don't think we need this method, we can just require within the test, it doesn't save us any lines.